### PR TITLE
chore: multiple licenses fixes

### PR DESCRIPTION
**Pull request by tisonkun**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Comments inline.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/meta-srv/src/failure_detector.rs
Comment (tisonkun) [inline src/meta-srv/src/failure_detector.rs:1]: Changes in this file refers to https://github.com/GreptimeTeam/greptimedb/pull/2709#discussion_r1386213829.

File: LICENSE [hunk 1]
```diff
@@ -186,7 +186,7 @@
       same "printed page" as the copyright notice for easier
       identification within third-party archives.
 
-   Copyright 2022 Greptime Team
+   Copyright [yyyy] [name of copyright owner]
```
Comment (tisonkun) [inline LICENSE:189]: Apache License 2.0 text is expected to use AS IS.

See also https://github.com/apache/incubator-answer/pull/527#discussion_r1325686233.

If you read this APPENDIX closely, you will know that here is a template.

File: licenserc.toml
Comment (tisonkun) [inline licenserc.toml:1]: See also https://github.com/apache/skywalking/issues/10223.

We should use a fixed `inceptionYear` instead of updating the license header every year.

File: licenserc.toml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+headerPath = "Apache-2.0.txt"
+
+includes = [
+    "*.rs",
+    "*.py",
+]
+
+[properties]
+inceptionYear = 2023
```
Comment (tisonkun) [inline licenserc.toml:23]: I guess it should be actually 2022. But it can introduce a large changeset so I'd prefer either do it in a separate commit or we just use 2023.

File: licenserc.toml
Comment (tisonkun) [reply licenserc.toml:1]: You can use

```
docker run --rm -v $(pwd):/github/workspace ghcr.io/korandoru/hawkeye-native:v3 format
```

from the project root to automatically apply the license header.

File: .github/workflows/license.yaml [hunk 3]
```diff
@@ -13,4 +13,4 @@ jobs:
     steps:
     - uses: actions/checkout@v2
     - name: Check License Header
-      uses: apache/skywalking-eyes/header@df70871af1a8109c9a5b1dc824faaf65246c5236
+      uses: korandoru/hawkeye@v3
```
Comment (waynexia) [inline .github/workflows/license.yaml:16]: What's the difference between these two apps? Like `skywalking-eyes` cannot add an empty line between license header and code?

File: licenserc.toml
Comment (waynexia) [reply licenserc.toml:1]: Could you please add this command to https://github.com/GreptimeTeam/greptimedb/blob/develop/CONTRIBUTING.md#before-pr ?

File: .github/workflows/license.yaml (hunk 3 above)
Comment (tisonkun) [reply .github/workflows/license.yaml:16]: Mainly for https://github.com/apache/skywalking/issues/10223 and I like my solution on hawkeye.

Other reasons are not quite strong but if you don't have Golang env, when you use the docker image to apply or check, hawkeye's image takes ~30MB while skywalking-eyes take > 500MB.

**Review by waynexia — APPROVED** [review]
LGTM, thanks @tisonkun

File: licenserc.toml
Comment (tisonkun) [reply licenserc.toml:1]: Updated.

**Review by MichaelScofield — APPROVED** [review]
